### PR TITLE
First pass on README updates, adds CONTRIBUTING file

### PR DESCRIPTION
**Pull request by jhaaaa**
- Adds all XMTP-generic content to README
- Provides placeholders for Dart-specific content
   - Please search on "dan" to find these placeholders. =)
   - And of course, you don't need to provide content for all placeholders - just ones that you are ready to provide and that you think will be useful for the Dev Preview. We can always iterate!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,188 @@
...
+
+This SDK is in **Developer Preview** status. We do **not** recommend using Developer Preview software in production apps.
+
+Software in this status:
+
+- Is not formally supported
+- Will change without warning
+- May not be backward compatible
+- Has not undergone a formal security audit
+
```
Comment (jhaaaa) [inline README.md:17]: Per feedback from Saul, let's update this text as follows:

```suggestion
This SDK is in **Developer Preview** status and ready for you to start building.

However, we do **not** recommend using Developer Preview software in production apps. Software in this status:

- Is not formally supported
- Will change without warning
- May not be backward compatible
- Has not undergone a formal security audit

```

**Review by dmccartney — APPROVED** [review]
Thank you for putting all of this together!

File: README.md [hunk 2]
```diff
@@ -1,4 +1,302 @@
...
-COMING SOON! This is a **work in progress** and should not be used yet. Follow along in the [tracking issue](https://github.com/xmtp/xmtp-flutter/issues/4) for updates on progress.
+![Test](https://github.com/xmtp/xmtp-flutter/actions/workflows/test.yml/badge.svg) ![Status](https://img.shields.io/badge/Project_Status-Developer_Preview-yellow)
+
+`xmtp-flutter` provides a Dart implementation of an XMTP message API client for use with Flutter apps.
+
+Use `xmtp-flutter` to build with XMTP to send messages between blockchain accounts, including DMs, notifications, announcements, and more.
+
+This SDK is in **Developer Preview** status and ready for you to start building.
+
+However, we do **not** recommend using Developer Preview software in production apps.
```
Comment (elisealix22) [inline README.md:11]: @jhaaaa @dmccartney not blocking, but was it intentional to remove this block:
```
However, we do not recommend using **Developer Preview software in production apps. Software in this status:
- Is not formally supported
- Will change without warning
- May not be backward compatible
- Has not undergone a formal security audit
```
Mostly asking for the security audit line!

**Review by elisealix22 — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (jhaaaa) [reply README.md:11]: @elisealix22 Thank you for asking! =) The edit was suggested by Yash, supported by Saul. <3 cc: @dmccartney

File: README.md [hunk 3]
```diff
@@ -171,52 +171,51 @@ var listening = client.streamMessages(convo).listen((message) {
...
-When sending a message you can specify the type of content. This allows you to specify different
-types of content than the default (a simple string, `ContentTypeText`). 
+When sending a message, you can specify the type of content. This allows you to specify different
+types of content than the default (a simple string, `ContentTypeText`).
 
-Support for other types of content can be added during client construction by registering additional
-including a `customCodecs` parameter. Every codec declares a specific content type identifier,
-`ContentTypeId` which is used to signal to the Client which codec should be used to process the 
-content that is being sent or received. See [XIP-5](https://github.com/xmtp/XIPs/blob/main/XIPs/xip-5-message-content-types.md) 
+Support for other types of content can be added during client construction by registering additional `ContentCodecs`, including a `customCodecs` parameter. Every codec declares a specific content type identifier,
```
Comment (jhaaaa) [inline README.md:211]: @dmccartney @elisealix22 There seemed to be a missing word between "additional" and "including" - so I added `ContentCodecs` based on https://github.com/xmtp/xmtp-js#different-types-of-content. 

Could you confirm that my edit is correct - or might you be able to provide the correction, please? =)

**Comment by jhaaaa** [comment]
@dmccartney and @elisealix22 - thank you for all of your help in getting this README done for today's release! AMAZING!

Made just a few style edits and left one question about the addition of `ContentCodecs` to a paragraph. 

[The diff makes it look like I made a bunch of changes - but it looks like my editor automatically (and unexpectedly) removed spaces at the end of a bunch of lines. I hope that this doesn't hurt anyone's sensibilities! <3]

File: README.md (hunk 3 above)
Comment (elisealix22) [reply README.md]: Updated & this should be resolved now!

**Pull request merged**